include/lsst/afw/image/ExposureInfo.h

```diff
@@ -344,6 +345,11 @@ class ExposureInfo final {
     std::shared_ptr<ApCorrMap> _apCorrMap;
     std::shared_ptr<image::VisitInfo const> _visitInfo;
     std::shared_ptr<TransmissionCurve const> _transmissionCurve;
+
+    // Class invariant: all values in _components are Storable or shared_ptr<Storable>
+    // Class invariant: if a key is absent from this object, the key may either
+    //     be absent from _components or may map to a null pointer.
```

Comment:
I don't necessarily object, but I'm curious why you decided to have two ways to represent an absent component.

Reply:
I think I was too tired to properly separate the interface and implementation. I'll switch to making `setComponent(nullptr)` remove the member.